### PR TITLE
Overhaul, working towards version 2.0

### PR DESCRIPTION
**Pull request by eXistenZNL**
### What has been done
- Switched to YML configuration
- Put the code in a class so it can be tested
- Possibility to set a different title and body per project
- PRs are now automatically assigned so they are no longer accidently overlooked
### Remarks

Before we release version 2.0 , we should test the PRBot class properly and update the README.md

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,3 @@
-.idea
+.idea/
 vendor/
-dev/config.php
+etc/config.yml
```
Comment (rick-nu) [inline .gitignore:3]: why etc?

File: bin/run.php [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        try {
+            $pullRequest = $prBot->createPR(
+                $fromBranch,
+                $toBranch,
+                $projectConfig['prTitle'],
+                $projectConfig['prBody']
+            );
+            $prBot->assignPR(
+                $pullRequest->getNumber(),
+                $projectConfig['assignees']
```
Comment (rick-nu) [inline bin/run.php:42]: Now assignees seem to be required alwasy? Or am I missing something

File: bin/run.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
+<?php
+
```
Comment (rick-nu) [inline bin/run.php:2]: This is personal preference, but I don't like empty lines at the top :innocent:

**Review by rick-nu — COMMENTED** [review]
Great improvements @eXistenZNL! I have some small feedback/suggestions we might want to consider before switching to master :)

File: composer.json [hunk 4]
```diff
@@ -2,8 +2,13 @@
...
     "description": "Automated pull requests for forgotten merge-backs",
     "require": {
+        "php": "^5.4",
+        "ext-curl": "*",
         "tan-tan-kanarek/github-php-client": "^1.0",
-        "ext-curl": "*"
+        "symfony/yaml": "^3.1"
+    },
+    "autoload": {
+        "psr-4": {"PRBot\\": "src/PRBot"}
```
Comment (rick-nu) [inline composer.json:11]: Why the extra `PRBot` folder? Seems very useless to me for this small library

File: .gitignore (hunk 1 above)
Comment (eXistenZNL) [reply .gitignore:3]: That´s where config files normally live under the [LSB](https://en.wikipedia.org/wiki/Linux_Standard_Base) :)

File: bin/run.php (hunk 2 above)
Comment (eXistenZNL) [reply bin/run.php:42]: You are right, I made a mental note of adding in a check for this indeed.

File: composer.json (hunk 4 above)
Comment (eXistenZNL) [reply composer.json:11]: True but I didn't know what do do with my life so I put it in an extra zakje.

File: bin/run.php (hunk 3 above)
Comment (eXistenZNL) [reply bin/run.php:2]: Meh, I do :(

File: composer.json (hunk 4 above)
Comment (rick-nu) [reply composer.json:11]: Lets remove the extra plastic then :wink:

**Pull request merged**